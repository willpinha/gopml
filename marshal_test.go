package base_test

import (
	"os"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
	base "github.com/willpinha/gopml"
)

func TestMain(m *testing.M) {
	v := m.Run()

	snaps.Clean(m)

	os.Exit(v)
}

func TestMarshal(t *testing.T) {
	o := base.Outline{
		Text: "Hello",
		Outlines: []base.Outline{
			{
				Text: "World",
			},
		},
	}

	data, err := o.Marshal()

	require.NoError(t, err)
	snaps.MatchSnapshot(t, string(data))
}
