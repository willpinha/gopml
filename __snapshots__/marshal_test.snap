
[TestMarshal - 1]
    <outline text="Hello">
        <outline text="World"></outline>
    </outline>
---
